packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -150,4 +151,45 @@ export class TagManager {
     }
     return true;
   }
+
+  private mergeFrom(propertyTags: any): void {
```

Comment:
I think I might be ready to refactor to some form a subclass or composition class for each type of tag. These switch statements are not clean in my opinion. Do you think it's worth it at this point?

Reply:
I totally think that's a good idea. Should there not be instances of `TagFormatter` that live on a taggable `Resource`?